apps/playground-web/src/components/headless-ui/token-examples.tsx

```diff
@@ -0,0 +1,245 @@
+"use client";
+
+import { THIRDWEB_CLIENT } from "@/lib/client";
+import { NATIVE_TOKEN_ADDRESS } from "thirdweb";
+import { ethereum } from "thirdweb/chains";
+import {
+  TokenIcon,
+  TokenName,
+  TokenProvider,
+  TokenSymbol,
+} from "thirdweb/react";
+import { CodeExample } from "../code/code-example";
+
+const USDC_ADDRESS = "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48";
+
+export function TokenImageBasic() {
+  return (
+    <>
+      <div className="space-y-2">
+        <h2 className="font-semibold text-2xl tracking-tight sm:text-3xl">
+          TokenIcon
+        </h2>
+        <p className="max-w-[600px] text-lg">
+          Show the default native icon of a network
+        </p>
+      </div>
+
+      <CodeExample
+        preview={
+          <TokenProvider
+            address={NATIVE_TOKEN_ADDRESS}
+            client={THIRDWEB_CLIENT}
+            chain={ethereum}
+          >
+            <TokenIcon className="h-auto w-20 rounded-full" />
+          </TokenProvider>
+        }
+        code={`import { TokenProvider, TokenIcon } from "thirdweb/react";
+
+function App() {
+  return (
+    <TokenProvider
+      address={NATIVE_TOKEN_ADDRESS}
+      client={THIRDWEB_CLIENT}
+      chain={ethereum}
+    >
+      <TokenIcon className="h-auto w-20 rounded-full" />
+    </TokenProvider>
+  )
+}`}
+        lang="tsx"
+      />
+    </>
+  );
+}
+
+export function TokenImageOverride() {
+  return (
+    <>
+      <div className="mt-8 space-y-2">
+        <h4 className="font-semibold text-lg">
+          Override the token&apos;s icon using the <b>iconResolver</b> prop.
+        </h4>
+        <p className="max-w-[600px] text-lg">
+          There is no official way to query the icon of a token. If you have a
+          source, you can pass it to the iconResolver prop.
+        </p>
+      </div>
+
+      <CodeExample
+        preview={
+          <TokenProvider
+            address={USDC_ADDRESS}
+            client={THIRDWEB_CLIENT}
+            chain={ethereum}
+          >
+            <TokenIcon
+              className="h-auto w-20 rounded-full"
+              iconResolver="/usdc.svg"
```

Comment:
is this the only way? what happens if i pass nothing?

i think we have all these icons in the pay service, could we just fetch those from there for tokens we know about?

Reply:
Oh I did not know we have the icons over there. I will check an update it in a follow up PR

Reply:
https://thirdwebdev.slack.com/archives/C04LA5X7FNE/p1733916180045149